elasticsearch/docker/dockerfiles/AL2.dockerfile

```diff
@@ -0,0 +1,73 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# This dockerfile generates an AmazonLinux-based image containing an OpenDistro for Elasticsearch (ODFE) installation.
+# It assumes that the working directory contains four files: an ODFE tarball (odfe.tgz), log4j2.properties, elasticsearch.yml, docker-entrypoint.sh.
+# Build arguments:
+#   ODFE_VERSION: Required. Used to label the image.
+#   BUILD_DATE: Required. Used to label the image. Should be in the form 'yyyy-mm-ddThh:mm:ssZ', i.e. a date-time from https://tools.ietf.org/html/rfc3339. The timestamp must be in UTC.
+#   UID: Optional. Specify the elasticsearch userid. Defaults to 1000.
+#   GID: Optional. Specify the elasticsearch groupid. Defaults to 1000.
+
+FROM amazonlinux:2 AS build
+
+ARG ODFE_VERSION
+ARG BUILD_DATE
+ARG UID=1000
+ARG GID=1000
+
+RUN yum install -y tar gzip shadow-utils
+
+# Create an elasticsearch user, group, and directory
+RUN groupadd -g $GID elasticsearch && \
```

Comment:
Can we save `elasticsearch` as variables like `UID`, `GID`?
So it is easier to change later on?

Reply:
I thought about adding that...but there's no demand for it right now, and it's easy to change later so I'd rather keep this change as small as possible.

Reply:
I'll add this in a new PR.